x/evidence/keeper/infraction.go

```diff
@@ -27,19 +27,6 @@ func (k Keeper) handleEquivocationEvidence(ctx sdk.Context, evidence *types.Equi
 	logger := k.Logger(ctx)
 	consAddr := evidence.GetConsensusAddress()
 
-	if _, err := k.slashingKeeper.GetPubkey(ctx, consAddr.Bytes()); err != nil {
-		// Ignore evidence that cannot be handled.
-		//
-		// NOTE: We used to panic with:
-		// `panic(fmt.Sprintf("Validator consensus-address %v not found", consAddr))`,
-		// but this couples the expectations of the app to both CometBFT and
-		// the simulator.  Both are expected to provide the full range of
-		// allowable but none of the disallowed evidence types.  Instead of
-		// getting this coordination right, it is easier to relax the
-		// constraints and ignore evidence that cannot be handled.
-		return
-	}
-
```

Comment:
If we do not panic l75, can we at least log something?

Reply:
Agree!

Reply:
Why don't we move the validator and pubkey check above this and we can short-circuit everything else?

Reply:
Perhaps we can replicate this no-op in the slashing [infractions.go ](https://github.com/cosmos/cosmos-sdk/blob/04750254669fdbe0e9909f962228a57023cd1fd2/x/slashing/keeper/infractions.go)?

Reply:
https://github.com/cosmos/cosmos-sdk/pull/15917

Why did this one just get closed? We asked both informal and SDK team for review.

Reply:
Your other PR did get closed because at the time it wasn't clear a change was needed. We double-checked and Informal as well verified. They came up with two possible solutions, and we asked them to submit a PR with the solution related to the SDK, so it is easier for us to review it.
Moreover, do note that PR targeting directly release branches gets closed anyway.

Reply:
Ok, I too, also prefer to focus entirely on the code. 

As long as you recognize the 1:1 nature of things here, we won't need to discuss further.  I apologize, prop 104 obligages us to report.

this is the last thing I want to spend time on.

Reply:
@julienrbrt I am quite sure informal did not review our detailed descriptions of why this is necessary, over the past month.

Reply:
@julienrbrt I am quite sure informal did not review our detailed descriptions of why this is necessary, over the past month.

Reply:
Again, I cannot speak for them and do not know what they did. However, I do not believe any team was ill intended, but probably just a communication issue.
(still marking all those comments as off-topic for readability, fyi)